core/ts/Poll.ts

```diff
@@ -161,18 +158,16 @@ export class Poll implements IPoll {
     this.stateTree = this.maciStateRef.stateTree.copy();
 
     // Create as many ballots as state leaves
-    const emptyBallot = new Ballot(this.maxValues.maxVoteOptions, this.treeDepths.voteOptionTreeDepth);
-    const emptyBallotHash = emptyBallot.hash();
-    this.ballotTree = new IncrementalQuinTree(this.stateTreeDepth, emptyBallot.hash(), STATE_TREE_ARITY, hash5);
-    this.ballotTree.insert(emptyBallotHash);
+    this.emptyBallotHash = this.emptyBallot.hash();
+    this.ballotTree = new IncrementalQuinTree(this.stateTreeDepth, this.emptyBallotHash, STATE_TREE_ARITY, hash5);
```

Comment:
Ok dumb question - why the need for a `ballotTree`? I thought there was just the state tree & message tree?

Reply:
The ballot tree, which will have the same amount of leaves as the state tree (1 Ballot <> 1 Signup), holds the computed ballot for a given MACI key. When we go and process messages, we update a voter's ballot depending on the content of the message processed, we hash it and include (replace) as leaf in the ballot tree.

Reply:
a public key which never casted a vote will have a empty ballot. All these ballots are then used in the tally process to tally the results

Reply:
Ok so a user's ballot is essentially their "final" vote? So it's their final valid message?

Reply:
There's no `ballotTree` in the contract state, correct? This is just for the off-chain processing/tallying of votes?

Reply:
Almost - each vote updates the ballot, so the ballot could be the result of 2 valid messages: a message is something like:

```json
{
  nonce: 1,
  voteOption: 1,
  voteWeight: 1,
  pubKey: PubKey1 
}
```

A ballot (at a high level) would be:

```json
[0, 0, 0, 0, 0, 0, 5]
```

Vote zero for option 1 to 6 and vote to option 7

Reply:
> There's no `ballotTree` in the contract state, correct? This is just for the off-chain processing/tallying of votes?

we only store a commitment to it (which is the hash of the root of the state treee, the root of the ballot tree and a salt)

Reply:
there could not be the same tree on chain because it is the result of the processing of the decrypted messages

Reply:
Awesome, appreciate the explanations 🙏 😄 

Ah, ok, because each command/message can only include one vote, right?

So e.g. in a QF/QV round, if I submit a ballot casting votes for 8 projects, I'm actually submitting 8 messages?

Reply:
Correct!